lib/surface/api.ex

```diff
@@ -381,7 +381,38 @@ defmodule Surface.API do
     end
   end
 
-  defp maybe_warn_mutually_exclusive_opts(_, _, _, _), do: nil
+  defp maybe_warn_mutually_exclusive_opts(:slot, :default, _, opts, caller) do
```

Comment:
this seemed the closest place we have currently for logging this type of warning; but I'm up for moving it if we think of a better place for this warning